internal/pkg/cli/job_delete.go

```diff
@@ -0,0 +1,206 @@
+)
+
+type deleteJobVars struct {
+	appName          string
+	skipConfirmation bool
+	name             string
+	envName          string
+}
+
+type deleteJobOpts struct {
+	deleteJobVars
+
+	// Interfaces to dependencies.
+	store  store
+	prompt prompter
+}
+
+func newDeleteJobOpts(vars deleteJobVars) (*deleteJobOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &deleteJobOpts{
+		deleteJobVars: vars,
+
+		store:  store,
+		prompt: prompt.New(),
+	}, nil
+}
+
+// Validate returns an error if the user inputs are invalid.
+func (o *deleteJobOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	if o.name != "" {
+		if _, err := o.store.GetJob(o.appName, o.name); err != nil {
+			return err
+		}
+	}
+	if o.envName != "" {
+		if err := o.validateEnvName(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Ask prompts the user for any required flags.
+func (o *deleteJobOpts) Ask() error {
+	if err := o.askJobName(); err != nil {
+		return err
+	}
+
+	if o.skipConfirmation {
+		return nil
+	}
+
+	// When there's no env name passed in, we'll completely
+	// remove the job from the application.
+	deletePrompt := fmt.Sprintf(fmtJobDeleteConfirmPrompt, o.name, o.appName)
+	deleteConfirmHelp := jobDeleteConfirmHelp
+	if o.envName != "" {
+		// When a customer provides a particular environment,
+		// we'll just delete the job from that environment -
+		// but keep it in the app.
+		deletePrompt = fmt.Sprintf(fmtJobDeleteFromEnvConfirmPrompt, o.name, o.envName)
+		deleteConfirmHelp = fmt.Sprintf(jobDeleteFromEnvConfirmHelp, o.envName)
+	}
+
+	deleteConfirmed, err := o.prompt.Confirm(
+		deletePrompt,
+		deleteConfirmHelp)
+
+	if err != nil {
+		return fmt.Errorf("job delete confirmation prompt: %w", err)
+	}
+	if !deleteConfirmed {
+		return errJobDeleteCancelled
+	}
+	return nil
+}
+
+func (o *deleteJobOpts) validateEnvName() error {
+	if _, err := o.targetEnv(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *deleteJobOpts) targetEnv() (*config.Environment, error) {
+	env, err := o.store.GetEnvironment(o.appName, o.envName)
+	if err != nil {
+		return nil, fmt.Errorf("get environment %s from config store: %w", o.envName, err)
+	}
+	return env, nil
+}
+
+func (o *deleteJobOpts) askJobName() error {
+	if o.name != "" {
+		return nil
+	}
+
+	names, err := o.jobNames()
+	if err != nil {
+		return err
+	}
+	if len(names) == 0 {
+		return fmt.Errorf("couldn't find any jobs in the application %s", o.appName)
+	}
+	if len(names) == 1 {
+		o.name = names[0]
+		log.Infof("Only found one job, defaulting to: %s\n", color.HighlightUserInput(o.name))
+		return nil
+	}
+	name, err := o.prompt.SelectOne(jobDeleteNamePrompt, "", names)
+	if err != nil {
+		return fmt.Errorf("select job to delete: %w", err)
+	}
+	o.name = name
+	return nil
+}
```

Comment:
Can we possibly use `o.sel.Job`? And could we update the counterpart in `svc delete` as well?

Reply:
☝️

Reply:
Yup!